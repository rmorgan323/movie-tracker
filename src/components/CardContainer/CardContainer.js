import React, { Component } from 'react';
// import './CardContainer.css';
import Card from '../Card/Card';
import { connect } from 'react-redux';

 const CardContainer = ( props ) => {
	
	const cards = props.movies.map((card) => {
		console.log(card)
		return (
			<Card 
				key={card.movieId}
				cardImage={card.image}
				cardTitle={card.title} />
		)
	})

	return (
		<div>
			{cards}
		</div>
	)
}

const mapStateToProps = (store) =>{
	return {
		movies: store.movies
	}
}

const mapDispatchToProps = (dispatch) =>{
	return {}
}


export default connect(mapStateToProps, null)(CardContainer)