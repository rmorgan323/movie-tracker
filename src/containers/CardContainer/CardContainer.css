.card-container {
  display: flex;
  flex-wrap: wrap;
  width: 1600px;
  margin: auto; }

@media screen and (max-width: 1599px) {
  .card-container {
    width: 1280px; } }

@media screen and (max-width: 1279px) {
  .card-container {
    width: 960px; } }

@media screen and (max-width: 959px) {
  .card-container {
    width: 640px; } }

@media screen and (max-width: 639px) {
  .card-container {
    width: 320px; } }
