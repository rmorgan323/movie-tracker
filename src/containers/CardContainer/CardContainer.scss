.card-container {
	display: flex;
	flex-wrap: wrap;
	width: 1600px;
	margin: auto;
}

.login-buttons {
	display: flex;
	flex-direction: row;
	justify-content: center;
}

@media screen and (max-width: 1599px) {
	.card-container {
		width: 1280px;
	}
}

@media screen and (max-width: 1279px) {
	.card-container {
		width: 960px;
	}
}

@media screen and (max-width: 959px) {
	.card-container {
		width: 640px;
	}
}

@media screen and (max-width: 639px) {
	.card-container {
		width: 320px;
	}
}

