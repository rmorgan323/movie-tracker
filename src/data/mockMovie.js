export const mockMovie = {
  userId: null,
  movieId: 141052,
  title: "Justice League",
  image: "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
  releaseDate: "2017-11-15",
  voteAverage: 6.6,
  overview: "Fueled by his restored faith in humanity and inspired",
};

export const mockMovieWithUser = {
  userId: 1,
  movieId: 141052,
  title: "Justice League",
  image: "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
  releaseDate: "2017-11-15",
  voteAverage: 6.6,
  overview: "Fueled by his restored faith in humanity and inspired",
};

export const mockMovieArray = [
	{
	  userId: 1,
	  movieId: 141052,
	  title: "Justice League",
	  image: "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
	  releaseDate: "2017-11-15",
	  voteAverage: 6.6,
	  overview: "Fueled by his restored faith in humanity and inspired",
	}
]

export const userWithMovieArray = {
	signedIn: true,
	userInfo: {
		id: 1,
		name: 'Batman',
		password: 'darkKn1ght',
		email: 'supermanSux@batmanrulez.com',
		favorites: mockMovieArray
	}
};