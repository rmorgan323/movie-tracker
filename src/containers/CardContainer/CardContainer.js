import React, { Component } from 'react';
import './CardContainer.css';
import Card from '../../components/Card/Card';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

export const CardContainer = props => {
  const cards = props.movies.map(card => {
    return (
      <Card
        key={card.movieId}
        userId={card.userId}
        id={card.movieId}
        title={card.title}
        image={card.image}
        releaseDate={card.releaseDate}
        voteAverage={card.voteAverage}
        overview={card.overview}
        favorite={card.favorite}
      />
    );
  });

  return <div className="card-container">{cards}</div>;
};

export const mapStateToProps = store => {
  return {
    movies: store.movies
  };
};

export default withRouter(connect(mapStateToProps, null)(CardContainer));
