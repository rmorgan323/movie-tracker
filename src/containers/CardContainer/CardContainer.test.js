import { CardContainer, mapStateToProps } from './CardContainer';
import React from 'react';
import { shallow } from 'enzyme';
import mockMovie from '../../data/mockMovie';
import mockUserData from '../../data/mockUser';

describe('CardContainer tests', () => {
  let mockMovies;
  let mockUser;
  let mockLocation;
  let renderedCardContainer;

  beforeEach(() => {
    mockMovies = [mockMovie];
    mockUser = mockUserData;
    mockLocation = {};

    renderedCardContainer = shallow(
      <CardContainer
        movies={mockMovies}
        user={mockUser}
        location={mockLocation}
      />
    );
  });
  it('Should match the snapshot', () => {
    expect(renderedCardContainer).toMatchSnapshot();
  });

  it('Should display cards from props', () => {
    expect(renderedCardContainer.find('Card').length).toEqual(1);
  });

  describe('mapStateToProps tests', () => {
    it('Should pull movies from store', () => {
      const mockStore = {
        movies: []
      };
      const result = mapStateToProps(mockStore);

      expect(result.movies).toEqual(mockStore.movies);
    });
  });
});
