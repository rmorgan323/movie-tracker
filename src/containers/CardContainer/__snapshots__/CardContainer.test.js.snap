// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    location={Object {}}
    movies={
        Array [
            Object {
              "favorite": false,
              "image": "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
              "movieId": 141052,
              "overview": "Fueled by his restored faith in humanity and inspired",
              "releaseDate": "2017-11-15",
              "title": "Justice League",
              "userId": null,
              "voteAverage": 6.6,
            },
          ]
    }
    user={
        Object {
            "signedIn": true,
            "userInfo": Object {
              "email": "",
              "favorites": Array [],
              "id": "",
              "name": "",
              "password": "",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Component)
          image="https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg"
          movieId={141052}
          overview="Fueled by his restored faith in humanity and inspired"
          releaseDate="2017-11-15"
          title="Justice League"
          userId=""
          voteAverage={6.6}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "141052",
        "nodeType": "class",
        "props": Object {
          "image": "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
          "movieId": 141052,
          "overview": "Fueled by his restored faith in humanity and inspired",
          "releaseDate": "2017-11-15",
          "title": "Justice League",
          "userId": "",
          "voteAverage": 6.6,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Component)
            image="https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg"
            movieId={141052}
            overview="Fueled by his restored faith in humanity and inspired"
            releaseDate="2017-11-15"
            title="Justice League"
            userId=""
            voteAverage={6.6}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "141052",
          "nodeType": "class",
          "props": Object {
            "image": "https://image.tmdb.org/t/p/w500/9rtrRGeRnL0JKtu9IMBWsmlmmZz.jpg",
            "movieId": 141052,
            "overview": "Fueled by his restored faith in humanity and inspired",
            "releaseDate": "2017-11-15",
            "title": "Justice League",
            "userId": "",
            "voteAverage": 6.6,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
